src/Sulu/Bundle/AdminBundle/Resources/js/containers/List/fieldTransformers/IconFieldTransformer.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import React from 'react';
+import type {Node} from 'react';
+import log from 'loglevel';
+import type {FieldTransformer} from '../types';
+import Icon from '../../../components/Icon';
+import iconFieldTransformerStyles from './iconFieldTransformer.scss';
+
+export default class IconFieldTransformer implements FieldTransformer {
+    transform(value: *, parameters: ?{[string]: mixed}): Node {
+        if (!value || !parameters) {
+            return value;
+        }
+
+        const {icons} = parameters;
+        if (!icons) {
+            return value;
+        }
+
+        if (typeof icons !== 'object') {
+            log.error('Parameter "icons" needs to be of type collection.');
+
+            return null;
+        }
+
+        const iconConfig = icons[value];
+        if (!iconConfig) {
+            return value;
+        }
+
+        let icon = null;
+        let color = null;
+        switch (typeof iconConfig) {
```

Comment:
Don't you think that code would be easier to read (and we could make more use of `const`) if we move that into two separate functions? Talking about something like that:

```javascript
if (typeof iconConfig === 'object) {
    return transformObjectConfig(iconConfig);
}

if (typeof iconConfig === 'string') {
    return transformStringConfig(iconConfig);
}

log.error('...');

return null;
```